src/shared/color-utils.ts

```diff
@@ -284,101 +307,129 @@ function setAlpha(color: string, value: number): string {
-    }
+    rgb = rgb.map((value) => {
+        const normalizedValue = value / 255;
 
-    return colorShadeMap;
+        return normalizedValue <= 0.039_28
+            ? normalizedValue / 12.92
+            : ((normalizedValue + 0.055) / 1.055) ** 2.4;
+    });
+
+    // Truncate at 3 digits
+    return Number((0.2126 * rgb[0] + 0.7152 * rgb[1] + 0.0722 * rgb[2]).toFixed(3));
 }
 
 /**
- * Converts any given valid and supported color string to rgb
- * @param {string} color - color string
- * @returns {string} rgb color string
+ * Darkens a color.
+ * @param {string} color - CSS color, i.e. one of: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla()
+ * @param {number} coefficient - multiplier in the range 0 - 1
+ * @returns {string} A CSS color string. Hex input values are returned as rgb
  */
-function convertToRgb(color: string): string {
-    const colorString = color.startsWith('hsl') ? hslToRgb(color) : color;
-    const decomposedColor = decomposeColor(colorString);
+function darken(color: string, coefficient: number): string {
+    const decomposedColor = decomposeColor(color);
+    const clampedCoefficient = Utils.clamp(coefficient);
+
+    if (decomposedColor.type.includes('hsl')) {
+        decomposedColor.values[2] *= 1 - clampedCoefficient;
+    } else if (decomposedColor.type.includes('rgb')) {
+        for (let index = 0; index < 3; index += 1) {
+            decomposedColor.values[index] *= 1 - clampedCoefficient;
+        }
+    }
 
     return recomposeColor(decomposedColor);
 }
 
 /**
- * Checks if a given string is a valid color string
- * @param {string} color - color string
- * @returns {boolean}
+ * Lightens a color.
+ * @param {string} color - CSS color, i.e. one of: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla()
+ * @param {number} coefficient - multiplier in the range 0 - 1
+ * @returns {string} A CSS color string. Hex input values are returned as rgb
  */
-function isValidColor(color: string): boolean {
-    const { type } = decomposeColor(color);
+function lighten(color: string, coefficient: number): string {
+    const decomposedColor = decomposeColor(color);
+    const clampedCoefficient = Utils.clamp(coefficient);
+
+    if (decomposedColor.type.includes('hsl')) {
+        decomposedColor.values[2] += (100 - decomposedColor.values[2]) * clampedCoefficient;
+    } else if (decomposedColor.type.includes('rgb')) {
+        for (let index = 0; index < 3; index += 1) {
+            decomposedColor.values[index] +=
+                (255 - decomposedColor.values[index]) * clampedCoefficient;
+        }
+    }
 
-    return supportedColorTypes.includes(type);
+    return recomposeColor(decomposedColor);
 }
 
-function blendColors(baseColor: string, layerColor: string): string {
-    const decomposedBase = decomposeColor(convertToRgb(baseColor));
-    const decomposedLayer = decomposeColor(convertToRgb(layerColor));
+/**
+ * Blends to colors in normal mode
```

Comment:
Thank you for the thorough documentation of these functions 👍

Reply:
not all credit goes to me on this ... most are copied and adjusted from the material ui functions ✨